File: Models/SQ_Soil_Temperature/src/cs/SQ_Soil_Temperature/SoilTemperatureExogenous.cs
using System;
using System.Collections.Generic;

public class SoilTemperatureExogenous 
{
    private double _meanTAir;
    private double _minTAir;
    private double _meanAnnualAirTemp;
    private double _maxTAir;
    private double _dayLength;
    
        public SoilTemperatureExogenous() { }
    
    
    public SoilTemperatureExogenous(SoilTemperatureExogenous toCopy, bool copyAll) // copy constructor 
    {
    if (copyAll)
    {
    
    meanTAir = toCopy.meanTAir;
    minTAir = toCopy.minTAir;
    meanAnnualAirTemp = toCopy.meanAnnualAirTemp;
    maxTAir = toCopy.maxTAir;
    dayLength = toCopy.dayLength;
    }
    }
    public double meanTAir
        {
            get { return this._meanTAir; }
            set { this._meanTAir= value; } 
        }
    public double minTAir
        {
            get { return this._minTAir; }
            set { this._minTAir= value; } 
        }
    public double meanAnnualAirTemp
        {
            get { return this._meanAnnualAirTemp; }
            set { this._meanAnnualAirTemp= value; } 
        }
    public double maxTAir
        {
            get { return this._maxTAir; }
            set { this._maxTAir= value; } 
        }
    public double dayLength
        {
            get { return this._dayLength; }
            set { this._dayLength= value; } 
        }
}

File: Models/SQ_Soil_Temperature/src/cpp/SQ_Soil_Temperature/SoilTemperatureExogenous.cpp
#include "SoilTemperatureExogenous.h"
using namespace SQ_Soil_Temperature;


SoilTemperatureExogenous::SoilTemperatureExogenous() {}

double SoilTemperatureExogenous::getmeanTAir() { return this->meanTAir; }
double SoilTemperatureExogenous::getminTAir() { return this->minTAir; }
double SoilTemperatureExogenous::getmeanAnnualAirTemp() { return this->meanAnnualAirTemp; }
double SoilTemperatureExogenous::getmaxTAir() { return this->maxTAir; }
double SoilTemperatureExogenous::getdayLength() { return this->dayLength; }

void SoilTemperatureExogenous::setmeanTAir(double _meanTAir) { this->meanTAir = _meanTAir; }
void SoilTemperatureExogenous::setminTAir(double _minTAir) { this->minTAir = _minTAir; }
void SoilTemperatureExogenous::setmeanAnnualAirTemp(double _meanAnnualAirTemp) { this->meanAnnualAirTemp = _meanAnnualAirTemp; }
void SoilTemperatureExogenous::setmaxTAir(double _maxTAir) { this->maxTAir = _maxTAir; }
void SoilTemperatureExogenous::setdayLength(double _dayLength) { this->dayLength = _dayLength; }